snapcraft/internal/parts.py

```diff
@@ -0,0 +1,92 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+
+import requests
+import yaml
+from progressbar import (ProgressBar, Percentage, Bar, AnimatedMarker)
+from xdg import BaseDirectory
+
+PARTS_URI = 'https://parts.snapcraft.io/v1/parts.yaml'
+
+logging.getLogger("urllib3").setLevel(logging.CRITICAL)
+logger = logging.getLogger(__name__)
+
+
+class _Base:
+
+    def __init__(self):
+        self.parts_dir = os.path.join(BaseDirectory.xdg_data_home, 'snapcraft')
```

Comment:
If the duplication is only in the tests I actually prefer that as we would detect the change